packages/story-editor/src/components/canvas/mediaRecordingLayer.js

```diff
@@ -0,0 +1,261 @@
+      'image/jpeg'
+    );
+    setFile(imageFile);
+  }, [setFile, setMediaBlobUrl]);
+
+  const debouncedCaptureImage = useDebouncedCallback(captureImage, 3000);
+
+  const onToggleVideoMode = useCallback(() => {
+    toggleIsGif();
+    if (videoRef.current) {
+      videoRef.current.play().catch(() => {});
+    }
+  }, [toggleIsGif]);
+
+  // Try to get permissions as soon as dialog opens.
+  useEffect(() => {
+    getMediaStream();
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- Only run once.
+  }, []);
+
+  useEffect(() => {
+    if (
+      previousAudioInput !== audioInput ||
+      previousVideoInput !== videoInput
+    ) {
+      getMediaStream();
+    }
+  }, [
+    audioInput,
+    getMediaStream,
+    previousAudioInput,
+    previousVideoInput,
+    videoInput,
+  ]);
+
+  useEffect(() => {
+    if (isReady) {
+      updateMediaDevices();
+    }
+  }, [isReady, updateMediaDevices]);
+
+  useEffect(() => {
+    // Checking for srcObject avoids flickering due to the stream changing constantly.
+    if (streamRef.current && !streamRef.current.srcObject && liveStream) {
+      streamRef.current.srcObject = liveStream;
+    }
+
+    if (streamRef.current && !liveStream) {
+      streamRef.current.srcObject = null;
+    }
+  }, [liveStream]);
+
+  if (needsPermissions) {
+    return <PermissionsDialog />;
+  }
+
+  if (isFailed) {
+    return <ErrorDialog />;
+  }
+
+  return (
+    <>
+      <LayerWithGrayout aria-label={__('Media Recording layer', 'web-stories')}>
+        <PageTitleArea>
+          <DurationIndicator />
+        </PageTitleArea>
+        <DisplayPageArea withSafezone={false} showOverflow>
+          <Wrapper>
+            {!isImageCapture && (
+              <>
+                {mediaBlobUrl && (
+                  <VideoMode value={!isGif} onChange={onToggleVideoMode} />
+                )}
+                <VideoWrapper>
+                  {mediaBlobUrl && (
+                    <>
+                      {/* eslint-disable-next-line styled-components-a11y/media-has-caption,jsx-a11y/media-has-caption -- We don't have tracks for this. */}
+                      <Video
+                        ref={videoRef}
+                        src={mediaBlobUrl}
+                        autoPlay
+                        controls={!isGif}
+                        muted={isMuted}
+                        loop={isGif}
+                        disablePictureInPicture
+                      />
+                    </>
+                  )}
+                  {!mediaBlob && liveStream && (
+                    <Video
+                      ref={streamRef}
+                      autoPlay
+                      muted
+                      disablePictureInPicture
+                    />
+                  )}
+                </VideoWrapper>
+              </>
+            )}
+            {isImageCapture && (
+              <VideoWrapper>
+                <Photo
```

Comment:
We could add decoding="async" here.

Reply:
It probably doesn't make any difference because it's basically an existing blob from the recording, but sure.